### DISCUSSION TITLE
123 Example Spreadsheet & Graphs

### DISCUSSION THREAD

**Discussion by rruhle** [taviso/123elf#111] [Show and tell]
Is this good place to share accounting spreadsheet developed on 123, and tested on 123elf ?

Lots of graph examples, 123 macro / functions, & methods of external file access.

**Comment by taviso** [comment]
Sure!

**Comment by rruhle** [comment]
Example Accounting spreadsheet
Test.zip unpacks to Test.wk3

Some explanations for poorly documented spreadsheet:

ALOT macro runs at startup & during bank views, from the VIEW menu, which checks for date-critical allotments (auto-withdrawals) and spits such transaction records to ARCHIVE22.csv (file name defined in SETUP from VIEW menu).

The Macro menu system is accessed from [Alt-M]
The SETUP section shows external file names, and lists [ALT] key assignments
Changing file names here is auto detected for dependency updates thru-out spreadsheet.

Some files names are automatically generated by "NEW YEAR" macro (found in POST menu). This worksheet is ready for New Year routine, since last updated in December 2022. Running NEW YEAR routine will backup old worksheet, and create some new file names.

The macro's are commented, some lines abandoned with "~" character, but old code saved to cell on right.  No further documentation available. Unless someone wants me to explain more, I did not anticipate anyone would adopt the whole system as is, without changing some things for themselves.

While exploring macro examples, be warned about long macro lines.

Spreadsheet Macro code can select different strings, depending how conditions change. Since macro strings change at run time, just looking at the current line displayed won't show all the options, without copying macro code to an editor, or using a STEP function that follows code execution & sub-routines, like 123 for DOS did.

Be warned that copying macro lines can also break all named reference pointers, unless a string character temporarily changes the entire line to static text, before re-locating on spreadsheet or to separate editor for better examination.

Best Regards,

Roger

[Test.zip](https://github.com/taviso/123elf/files/10339248/Test.zip)

**Comment by rruhle** [comment]
For those wondering why anyone would use legacy software for bookkeeping?

1) In the 1980's & 90's, the legacy Lotus 1-2-3 natural macro language was easy for beginners working with spreadsheets.
2) Legacy DOS apps were not broken by modern OS updates that forced consumers to buy app upgrades to restore data.
3) Most computer viruses, malware, & hackers ignore obsolete files for DOS, much less obsolete & unheard of for Unix.
4) By requiring records proof, rather than GAAP, TAX law is agnostic on bookkeeping, ledger, or profit & loss methods.
5) Any spreadsheet can simplify record keeping, from a checkbook register, or complete business records & reports from CSV files.

The attached Test.zip bookkeeping spreadsheet shows all Allotments coming from the same credit card reconciliation "CARD2", and other bill-payments setup from a checking account reconciliation "BANK2". For me it easier to fix errors during bank reconciliations with separate deposit and expense accounts. Others may wish to combine everything on 1 bank statement.
